tests/lib/tools/fs-state

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: fs-state wait-for-file [filename] [iterations] [sleep-time]"
+    echo "usage: fs-state mock-dir [dirname]"
+    echo "usage: fs-state restore-dir [dirname]"
+    echo "usage: fs-state mock-file [filename]"
+    echo "usage: fs-state restore-file [filename]"
+}
+
+wait_for_file() {
+    local file="$1"
+    local iters="$2"
+    local sleep_time="$3"
+
+    for _ in $(seq "$iters"); do
+        if [ -e "$file" ]; then
+            echo "File exists: $file"
+            return 0
+        fi
+        sleep "$sleep_time"
+    done
+    echo "File does not exist: $file"
+    return 1
+}
+
+_mock_dir() {
+    local dir="$1"
+    if ! [ -d "$dir" ]; then
+        echo "Creating dir: $dir"
+        mkdir -p "$dir"
+        touch "$dir.fake"
+    fi
+}
+
+mock_dir() {
+    local dir="$1"
+    if [ -d "$dir" ]; then
+        mv "$dir" "$dir.back"
+    fi
+    _mock_dir "$dir"
+}
+
+restore_dir() {
+    local dir="$1"
+
+    echo "Cleaning dir: $dir"
+    if [ -e "$dir.fake" ]; then
+        rm -rf "$dir"
+        rm -f "$dir.fake"
+    else
+        echo "The directory $dir is not mocked"
+        exit 1
+    fi
+    if [ -d "$dir.back" ]; then
+        mv "$dir.back" "$dir"
+    fi
+}
+
+_mock_file() {
+    local file="$1"
+    if ! [ -e "$file" ]; then
+        echo "Creating file: $file"
+        echo "content for $file" > "$file"
+        echo "content for fake $file" > "$file.fake"
```

Comment:
Is the text in the fake file relevant? Perhaps it could say something like this:
```
The .fake file is a marker file used by fs-state.
```

Reply:
done